docs/byoh_agent.md

```diff
@@ -0,0 +1,121 @@
+# BYOH Agent
+
+BYOH agent is a binary that runs on the hosts and its responsibility include -  
+1. Setup - Install kubernetes components on the host
+2. Registration - Register the host to the cluster capacity pool
+3. Bootstrap - Convert the host to a kubernetes node
+
+## Usage of BYOH agent
+
+Below flags are supported by the agent:-  
+```
+--downloadpath string 
+```
+File System path to keep the downloads (default `/var/lib/byoh/bundles`)  
+<br />
+
+```
+--feature-gates map[String]Bool
+```
+A set of key-value pairs that describe feature gates for alpha/experimental features.  
+<br />
+```
+--kubeconfig string           
+```
+Paths to a kubeconfig. Only required if out-of-cluster.  
+<br />
+```
+--label labelFlags       
+```
+labels to attach to the ByoHost CR in the form `labelname=labelVal` Eg: `--label site=apac --label cores=2`  
+<br />
+```
+--metricsbindaddress string
+```
+metricsbindaddress is the TCP address that the controller should bind to for serving prometheus metrics.It can be set to "0" to disable the metrics serving (default ":8080")  
+<br />
+```
+--namespace string
+```
+Namespace in the management cluster where you would like to register this host (default "default")  
+<br />
+```
+--skip-installation
+```
+If you want to skip installation of the kubernetes component binaries  
```

Comment:
Add a note saying it will be user's responsibility to manage kubernetes components in that case